src/function.rs

```diff
@@ -278,37 +301,122 @@ where
   }
 }
 
+pub struct FunctionBuilder<'s> {
+  callback: FunctionCallback,
+  data: Option<Local<'s, Value>>,
+  length: i32,
+  constructor_behavior: ConstructorBehavior,
+  side_effect_type: SideEffectType,
+  name: Option<Local<'s, String>>,
+}
+
+impl<'s> FunctionBuilder<'s> {
+  fn new(callback: impl MapFnTo<FunctionCallback>) -> FunctionBuilder<'s> {
+    FunctionBuilder {
+      callback: callback.map_fn_to(),
+      data: None,
+      length: 0,
+      constructor_behavior: ConstructorBehavior::Allow,
+      side_effect_type: SideEffectType::HasSideEffect,
```

Comment:
One downside is that it gets out of sync with v8.h if upstream decides to change the defaults.

Not a criticism because I don't know of a good way to prevent that. Tests may catch changes to `ConstructorBehavior` but `SideEffectType` is a lot harder to observe directly.

Reply:
yeah I noticed we do this for PromiseState already. I was thinking after this lands I might set up bindgen so we automagically use v8's values